### DISCUSSION TITLE
[RFC] ConfigProvider support custom sub components

### DISCUSSION THREAD

**Discussion by MadCcc** [ant-design/ant-design#46170] [RFCs]
## Motivation

https://github.com/ant-design/ant-design/issues/45893

## API

```tsx
<ConfigProvider textArea={{ className: 'class', style: {} }}>
  <Input.TextArea />
</ConfigProvider>
```

## Components

Completely seperated component
- [ ] Badge.Ribbon
- [x] Input.TextArea
- [ ] ~~Tag.CheckableTag~~ (`checkable` prop)
- [ ] Space.Compact

Group (`classNames.group`)
- [ ] Avatar.Group
- [ ] Checkbox.Group
- [ ] FloatButton.Group
- [ ] Radio.Group

Wrap (pending)
- [ ] Alert.ErrorBoundary
- [ ] Cascader.Panel
- [ ] FloatButton.BackTop
- [ ] Input.Password
- [ ] Input.Search
- [ ] Radio.Button
- [ ] Tree.DirectoryTree
- [ ] Typography.Link
- [ ] Typography.Text
- [ ] Typography.Paragraph
- [ ] Typography.Title

**Comment by li-jia-nan** [comment]
会不会有歧义？下面这样会不会更好一点？

```tsx
<ConfigProvider input={{ textArea: { className: 'class', style: {} }}}>
  <Input.TextArea />
</ConfigProvider>
```

**Reply by MadCcc** [reply]
textArea 不是 input 的属性，就是因为这个才需要单独拿出来
